gb-roms/src/controllers/mbc1.rs

```diff
@@ -0,0 +1,318 @@
+use crate::header::size::{RamSize, RomSize};
+use gb_cpu::{address_bus::Error, FileOperation, Position, RomOperation};
+use std::io::{self, Read};
+
+pub const MBC1_ROM_SIZE: usize = 0x4000;
+pub const MBC1_MAX_ROM_BANK: usize = 0x80;
+pub const MBC1_RAM_SIZE: usize = 0x2000;
+pub const MBC1_MAX_RAM_BANK: usize = 0x4;
+
+pub struct MBC1 {
+    configuration: Configuration,
+    rom_bank: Vec<[u8; MBC1_ROM_SIZE]>,
+    ram_bank: Vec<[u8; MBC1_RAM_SIZE]>,
+    regs: MBC1Reg,
+}
+
+impl MBC1 {
+    pub fn new(ram_size: RamSize, rom_size: RomSize) -> Self {
+        let ram_bank = ram_size.get_bank_amounts();
+        let rom_bank = rom_size.get_bank_amounts();
+
+        Self {
+            configuration: Configuration::from_sizes(ram_size, rom_size),
+            rom_bank: vec![[0_u8; MBC1_ROM_SIZE]; rom_bank],
+            ram_bank: vec![[0_u8; MBC1_RAM_SIZE]; ram_bank],
+            regs: MBC1Reg::default(),
+        }
+    }
+
+    pub fn from_file(
```

Comment:
As in my previous comment, I think file reading is out of scope for a controller

Reply:
It's a choice I've made, If you've a better way to initialise the controller you can do a PR ;)